lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -1256,6 +1257,69 @@ struct AssertLikeOpConversion : public OpConversionPattern<MooreOpTy> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// Format String Conversion
+//===----------------------------------------------------------------------===//
+
+struct FormatLiteralOpConversion : public OpConversionPattern<FormatLiteralOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(FormatLiteralOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<sim::FormatLitOp>(op, adaptor.getLiteral());
+    return success();
+  }
+};
+
+struct FormatConcatOpConversion : public OpConversionPattern<FormatConcatOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(FormatConcatOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<sim::FormatStringConcatOp>(op,
+                                                           adaptor.getInputs());
+    return success();
+  }
+};
+
+struct FormatIntOpConversion : public OpConversionPattern<FormatIntOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(FormatIntOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // TODO: These should honor the width, alignment, and padding.
+    switch (op.getFormat()) {
+    case IntFormat::Decimal:
+      rewriter.replaceOpWithNewOp<sim::FormatDecOp>(op, adaptor.getValue());
```

Comment:
Do we have a way of handling signedness here? By not setting the `signed` attribute the value will always get formatted as unsigned. How would we deal with the difference in this example:
```SV
module tb;
  byte foo = 8'h80;
  initial $display("Signed:   %d", $signed(foo));
  initial $display("Unsigned: %d", $unsigned(foo));
endmodule
```

Reply:
That's a great point. Do you know what the standard says how your example would be printed? Does `%d` check the type's signedness?

Reply:
I don't see any mention of the sign in the LRM. But for once all simulators on edaplayground agree: It depends on the type's signedness. The above example is the same as
```SV
module tb;
  logic signed [7:0] fooS = 8'h80;
  logic [7:0] fooU = 8'h80;
  initial $display("Signed:   %d", fooS); // "Signed:   -128"
  initial $display("Unsigned: %d", fooU); // "Unsigned: 128"
endmodule
```
I wish they'd also agree on the padding...

Reply:
🎉 Nice! I also love how the `%d` field expands to the maximum width a decimal number can have, 3, but then adds a random `-` in front that throws everything off 🤣. I love how SV consistently picks the wrong default for a lot of things.